lib/bridge.js

```diff
@@ -237,11 +240,109 @@ Bridge.prototype.run = function(port, config, appServiceInstance) {
         }
         self.opts.controller.onLog(line, false);
     });
+    this.customiseAppservice(this.appService);
     this.appService.listen(port);
     return this.loadDatabases();
 };
 
 /**
+ * Apply any customisations required on the appService object.
+ * Subclasses are intended to override this method where appropriate. If
+ * overridden the subclass method will still have to call the one in this
+ * class.
```

Comment:
The IRC bridge currently does something similar to this in order to do linking/unlinking of rooms for plumbing. If this were part of the public API, we could do it properly without having to gut wrench like we are currently doing (and will do in all other bridges like the Slack bridge).